### PR TITLE
UCP/RNDV: Handle failure in rndv_get operation

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

File: src/ucp/tag/rndv.c [hunk 1]
```diff
@@ -303,23 +303,27 @@ static void ucp_rndv_zcopy_recv_req_complete(ucp_request_t *req, ucs_status_t st
...
     UCS_PROFILE_REQUEST_EVENT(rreq, "complete_rndv_get", 0);
 
     ucp_rkey_destroy(rndv_req->send.rndv_get.rkey);
     ucp_request_send_buffer_dereg(rndv_req);
 
-    ucp_rndv_req_send_ats(rndv_req, rreq, rndv_req->send.rndv_get.remote_request,
-                          UCS_OK);
-    ucp_rndv_zcopy_recv_req_complete(rreq, UCS_OK);
+    if (status == UCS_OK) {
+        ucp_rndv_req_send_ats(rndv_req, rreq,
```
Comment (dmitrygx) [inline src/ucp/tag/rndv.c:323]: why do not send the bad status to the sender in case of error as we do for RKEY_PTR ?

Comment (yosefe) [reply src/ucp/tag/rndv.c:323]: we cannot send anything to sender if the connection is failed

Comment (dmitrygx) [reply src/ucp/tag/rndv.c:323]: @yosefe I see

File: src/ucp/tag/rndv.c [hunk 2]
```diff
@@ -559,7 +563,7 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_get_completion, (self, status),
                                                send.state.uct_comp);
 
     if (rndv_req->send.state.dt.offset == rndv_req->send.length) {
-        ucp_rndv_complete_rma_get_zcopy(rndv_req);
+        ucp_rndv_complete_rma_get_zcopy(rndv_req, status);
```
Comment (brminich) [inline src/ucp/tag/rndv.c:566]: should not we call `ucp_rndv_complete_rma_get_zcopy` even if `dt.offset != send.length`, but status is not ok?

**Comment by yosefe** [comment]
bot:pipe:retest

**Comment by yosefe** [comment]
bot:pipe:retest

File: src/ucp/tag/rndv.c (hunk 2 above)
Comment (yosefe) [reply src/ucp/tag/rndv.c:566]: in this case it should be completed from pending queue , by ucp_ep_err_pending_purge

Comment (yosefe) [reply src/ucp/tag/rndv.c:566]: @brminich FYI

Comment (brminich) [reply src/ucp/tag/rndv.c:566]: ok

**Pull request merged**